ra2ce/analysis/adaptation/adaptation_option.py

```diff
@@ -121,30 +121,25 @@ def calculate_cost(self, time_horizon: float, discount_rate: float) -> float:
         def calc_years(from_year: float, to_year: float, interval: float) -> range:
             return range(
                 round(from_year),
-                round(min(to_year, time_horizon)),
+                round(min(to_year, time_horizon))+1,
                 round(interval),
             )
 
         def calc_cost(cost: float, year: float) -> float:
-            return cost * (1 - discount_rate) ** year
+            return cost / (1 + discount_rate) ** year
 
-        _constr_years = calc_years(
-            0,
+        _assessment_years = calc_years(
+            1,
             time_horizon,
-            self.construction_interval,
+            1,
         )
-        _lifetime_cost = 0.0
-        for _constr_year in _constr_years:
-            # Calculate the present value of the construction cost
-            _lifetime_cost += calc_cost(self.construction_cost, _constr_year)
-
-            # Calculate the present value of the maintenance cost
-            _maint_years = calc_years(
-                _constr_year + self.maintenance_interval,
-                _constr_year + self.construction_interval,
-                self.maintenance_interval,
-            )
-            for _maint_year in _maint_years:
-                _lifetime_cost += calc_cost(self.maintenance_cost, _maint_year)
+        _lifetime_cost = self.construction_cost
+        for _constr_year in _assessment_years:
+
+            if self.construction_interval > 0 and _constr_year in range(0, time_horizon, round(self.construction_interval)):
+                _lifetime_cost += calc_cost(self.construction_cost, _constr_year)
+
+            if self.maintenance_interval > 0 and _constr_year in range(0, time_horizon + 1, round(self.maintenance_interval)): # +1 here need but no above
+                _lifetime_cost += calc_cost(self.maintenance_cost, _constr_year)
```

Comment:
What is the reason for this change? Does it have impact on the outcome? Can `construction_interval` be 0?
Imo the readability is less.
Why evaluate each year?
`calc_year` doesn't seem to be used anymore.

Reply:
Reason is to allow construction_interval to be 0.
Imo this proposition is clearer:
- 5 lines shorters
- no double loops
- When i was debugging, it was easier to look at costs year after year.